### DISCUSSION TITLE
Using i18n

### DISCUSSION THREAD

**Discussion by AndresDiazG** [PanJiaChen/vue-element-admin#3939] [Q&A]
May I ask how to use the i18n component we all see on the demo project? I liked the view and I can see a lot of uses of it in my website but I can't find it on the Master branch or in any other branch this project has.

Thanks

AD.

**Comment by markhermano** [comment]
Hi

In order to use i18n, you need to add translations in the language files in this dir `/src/lang/`.
For example:
If I have these translations setup.
```js
//en.js English translation
export default {
  response: {
    accept: 'yes',
    deny: 'no',
    doubt: 'maybe'
  }
}
```
```js
//es.js Spanish translation
export default {
  response: {
    accept: 'sí',
    deny: 'no',
    doubt: 'quizás'
  }
}
```

then I can access them in vue template by:
```vue
<template>
  <div>
    <ul>
      <li>{{ $t('response.accept') }}</li>
      <li>{{ $t('response.deny') }}</li>  
      <li>{{ $t('response.doubt') }}</li>  
    </ul>
  </div>
</template>
<script>
  export default {
    mounted() {
      console.log(this.$t('response.accept'))
    }
  }
</script>
```

Make sure to have i18n imported in the index.js file
```js
import i18n from './lang'
```

In a js file, I can access it by:
```js
import i18n from '~/lang'

const accept = i18n.t('response.accept')

export default accept
```
